resource_detectors/lib/opentelemetry/resource/detectors/deployment.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module OpenTelemetry
+  module Resource
+    module Detectors
+      # Deployment contains detect class method for determining deployment resource attributes
+      module Deployment
+        extend self
+
+        def detect
+          resource_attributes = {}
+          deployment_environment = rails_env || sinatra_env || rack_env
+          resource_attributes[::OpenTelemetry::SemanticConventions::Resource::DEPLOYMENT_ENVIRONMENT] = deployment_environment if deployment_environment
+          OpenTelemetry::SDK::Resources::Resource.create(resource_attributes)
+        end
+
+        private
+
+        def rails_env
+          # rails extract env like this:
+          # https://github.com/rails/rails/blob/5647a9c1ced68d20338552d47a3b755e10a271c4/railties/lib/rails.rb#L74
+          # ActiveSupport::EnvironmentInquirer.new(ENV["RAILS_ENV"].presence || ENV["RACK_ENV"].presence || "development")
+          ::Rails.env.to_str if defined?(::Rails.env)
```

Comment:
I was actually starting with just checking the `::Rails` and then something crashed in my playground application when verifying that the detector is working.
Will try to reproduce it and paste here an example so we can analyse it. Is it problematic in any way to check for `defined?(::Rails.env)` instead of `defined?(::Rails)`?

Reply:
after some testing it seems fine to check `::Rails.env`. 

Would be curious to know what the error was. I just modified my suggestion above, i think we want to be invoking `.to_s` and not `.to_str` here, as there is a difference. `.to_str` returns a "string like object" while `.to_s` returns a string representation of the object. 

Frankly I'm not sure if we even need to invoke `.to_s` at all, since afaik `Rails.env` ought to return a string, but if there's any doubt then `.to_s` would be appropriate (and `.to_str` would likely be incorrect)

Reply:
> i think we want to be invoking .to_s and not .to_str here, as there is a difference. .to_str returns a "string like object" while .to_s returns a string representation of the object.

Thanks for spotting that, I wasn't aware of the difference, was just trying `to_str` and it worked. Changed to use `to_s`

> Frankly I'm not sure if we even need to invoke .to_s at all, since afaik Rails.env ought to return a string, but if there's any doubt then .to_s would be appropriate (and .to_str would likely be incorrect)

The value that is [returned from rails env function is: `ActiveSupport::EnvironmentInquirer`](https://github.com/rails/rails/blob/5647a9c1ced68d20338552d47a3b755e10a271c4/railties/lib/rails.rb#L74).
From the docs:

```
  # Wrapping a string in this class gives you a prettier way to test
  # for equality. The value returned by <tt>Rails.env</tt> is wrapped
  # in a StringInquirer object, so instead of calling this:
  #
  #   Rails.env == 'production'
  #
  # you can call this:
  #
  #   Rails.env.production?
  #
```

This is the reason `to_s` is mandatory for the env value before we store it into the resource

Reply:
> Would be curious to know what the error was

So I wasn't able to reproduce it in a real setup, it works good with both `defined?(::Rails.env)` and `defined?(::Rails)`, and also for sinatra.
But if I remove the `env` from the check, current test suite fails with things like:
```
  1) Error:
OpenTelemetry::Resource::Detectors::Deployment::.detect#test_0001_returns an empty resource:
NoMethodError: undefined method `env' for Rails:Module
Did you mean?  end
    /Users/amirblum/repos/opentelemetry-ruby/resource_detectors/lib/opentelemetry/resource/detectors/deployment.rb:23:in `rails_env'
    /Users/amirblum/repos/opentelemetry-ruby/resource_detectors/lib/opentelemetry/resource/detectors/deployment.rb:12:in `detect'
    /Users/amirblum/repos/opentelemetry-ruby/resource_detectors/test/opentelemetry/detectors/deployment_test.rb:9:in `block (3 levels) in <top (required)>'
```

Any thought on this? should I update the tests somehow or just leave it the way it is?